src/libraries/System.Private.CoreLib/src/System/IO/Strategies/Net5CompatFileStreamStrategy.Unix.cs

```diff
@@ -13,9 +13,6 @@ namespace System.IO.Strategies
     /// <summary>Provides an implementation of a file stream for Unix files.</summary>
     internal sealed partial class Net5CompatFileStreamStrategy : FileStreamStrategy
```

Comment:
Unrelated to this PR, but I was surprised to see we didn't have a new strategy for unix, one that reused the same buffering strategy, one that didn't have "Net5Compat" in its name, etc.  I get we've not yet done the io_uring implementation, but presumably we can still do a tad better than this one?

Reply:
We can do better on Unix, but all the major bugs were related to Windows implementation and that is why we have not introduced a new Unix strategy.

This will be mandatory for removing the `Net5Compat*` things when we snap for .NET 7.

With the new `RandomAccess` APIs introduced in this PR it should be relatively easy to implement a new Unix strategy (we could have one offset-based for seekable files and one for non-seekable files).  @stephentoub Do you think that this should ship as part of .NET 6?

Reply:
> Do you think that this should ship as part of .NET 6?

That would be my preference.

Reply:
I'll see what I can do, but I can't promise anything (tight schedule)

Reply:
Created an issue to keep track of it: https://github.com/dotnet/runtime/issues/53809